File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/storage/TripsStorage.java
package com.worldventures.dreamtrips.modules.trips.storage;

import com.worldventures.janet.cache.CachedAction;
import com.worldventures.janet.cache.storage.ActionStorage;
import com.worldventures.janet.cache.storage.CombinedListStorage;
import com.worldventures.janet.cache.storage.Storage;
import com.worldventures.dreamtrips.modules.trips.command.GetTripsCommand;
import com.worldventures.dreamtrips.modules.trips.model.TripModel;

import java.util.List;

public class TripsStorage extends CombinedListStorage<TripModel> implements ActionStorage<List<TripModel>> {

   public TripsStorage(Storage<List<TripModel>> memoryStorage, Storage<List<TripModel>> diskStorage) {
      super(memoryStorage, diskStorage);
   }

   @Override
   public Class<? extends CachedAction> getActionClass() {
      return GetTripsCommand.class;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/storage/TripsStorage.kt
package com.worldventures.dreamtrips.modules.trips.service.storage

import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsCommand
import com.worldventures.dreamtrips.modules.trips.model.TripModel
import com.worldventures.janet.cache.storage.ActionStorage
import com.worldventures.janet.cache.storage.CombinedListStorage
import com.worldventures.janet.cache.storage.Storage

class TripsStorage(memoryStorage: Storage<List<TripModel>>, diskStorage: Storage<List<TripModel>>) :
      CombinedListStorage<TripModel>(memoryStorage, diskStorage), ActionStorage<List<TripModel>> {

   override fun getActionClass() = GetTripsCommand::class.java
}

